### ISSUE TITLE
[BUG] Wrong selection in side menu after clicking an entry

### ISSUE THREAD

**Issue by simonnagl** [firehol/netdata#2356]
Version: v1.6.0-553-g0d75bc6b_rolling
Browser: Safari Version 10.1.1 (12603.2.4)

Problem:
After clicking an entry in the Side menu the wrong menu item gets selected. (see Screenshot)

<img width="1362" alt="bildschirmfoto 2017-06-18 um 10 51 11" src="https://user-images.githubusercontent.com/7253893/27259464-28ffb33c-5414-11e7-9b12-59bac5b8e7ad.png">

Memory was clicked and page jump works but the item above (processes) gets the selected menu item. This issue appears with all items.

**Comment by ktsaou** [comment]
I think this is related to screen resolution.
If you resize the window vertically it is ok.

I can't reproduce it, but I have seen it once.
So, can you try changing this: https://github.com/ktsaou/netdata/blob/06b713400b16279ad2b3780c07380e328a374c7d/web/index.html#L2733-L2736

with this:

```js
            var scrollspyOffset = $(window).height() / 5;
            if(scrollspyOffset > 200) scrollspyOffset = 200;
            if(scrollspyOffset < 50) scrollspyOffset = 50;
            $(document.body).scrollspy({
                target: '#sidebar',
                offset: scrollspyOffset // controls the diff of the <hX> element to the top, to select it
            });
```

**Comment by ktsaou** [comment]
If it works for you, I'll merge it.

**Comment by ktsaou** [comment]
merged the fix.
Check it. If it still does not work, post here to find another way.

**Comment by simonnagl** [comment]
Thank you for the quick response.

Same error on v1.6.0-578-g7de06bae_rolling

Nothing changed for me. Resizing to any width or height does not change anything. I'll not yet tried to change resolution of my screen. The menu gets the right selection if I scroll ca 28px down.

Tell me if I can help reproducing this. On chrome the position is much higher.

Please tell me if I can do something to help you. PS: You do not need to fix this just now. For me this is a minor bug.

**Comment by ktsaou** [comment]
It is annoying for sure.
Are you sure your browser got the latest version of index.html? Press Control-F5.

If the right version is loaded, try to play a bit with the thresholds (50 to 200). Try setting them to 0, 50, 100, 200. Does anything change?

**Comment by simonnagl** [comment]
Very strange behavior now. It sometimes works and sometimes not. Also for the actual code.

If I refresh the page sometimes no item in the sidebar is selected. If this is the case then scrolling or clicking the sidebar is buggy as described above.

Sometimes one item in the sidebar gets selected (the right one). Then netdata just works as expected.

Which of the two scenarios appear seems totally random. Clicking F5 again and again and doing nothing else...

I also checked if switching resolution has any effect but no.

**Comment by ktsaou** [comment]
strange indeed. Did you try to change the thresholds?

**Comment by simonnagl** [comment]
Yes. This does not change anything on the random loading issue. The configured offsets seem OK when loading correctly. Changing offset works like expected and in other browsers.

**Comment by ktsaou** [comment]
When you have this random issue, could you please check if the browser reports that `index.html` is loaded from netdata or its cache? You may have a "dirty" cache that somehow gets loaded, instead of loading the latest netdata version.

**Comment by simonnagl** [comment]
Checked. This is not the fact. I also disabled browser caching. Same random issue...

**Comment by simonnagl** [comment]
@ktsaou: Could you please reopen the bug? Maybe someone can also reproduce it and find a solution.

**Comment by ktsaou** [comment]
of course...
Unfortunately I can't reproduce it. I guess that the solution would be pretty simple if a method to reproduce it is found.

**Comment by ktsaou** [comment]
found it and fix it, in PR #3033

**Comment by simonnagl** [comment]
I could just verify the fix. Thank you! You are doing an awesome job!
